### PR TITLE
Robyn one pager: Ad stock and immediate_vs_carryover implementation

### PR DESCRIPTION
**Pull request by sumalreddy17**
immediate vs carryover response visualization

Implements stacked bar chart comparing immediate and carryover effects
Displays percentage response contributions with centered labels
Uses consistent color scheme (#59B3D2 for immediate, coral for carryover)
Formats axis with percentages
Positions legend at top with two columns
Matches R implementation styling and functionality
Supports both standalone and subplot integration

Part of Marketing Mix Modeling visualization suite.

Testing:
Screenshot 2024-11-06 at 11 31 20 AM

Graph plotted by R:
Screenshot 2024-11-06 at 2 53 54 PM

adstock rate visualization

Implements dual-mode adstock rate visualization:

Geometric: horizontal bar chart with percentage labels
Weibull: multi-panel decay curve plots with halflife indicators

Handles both CDF and PDF Weibull variants
Adapts layout based on number of channels
Formats axes with percentages and proper intervals
Matches R implementation styling
Supports both standalone and subplot integration

Testing:

Screenshot 2024-11-06 at 9 54 45 AM

Graph plotted by R:
Screenshot 2024-11-06 at 2 53 48 PM

### CODE REVIEW

File: python/src/robyn/visualization/pareto_visualizer.py [hunk 1]
```diff
@@ -234,23 +234,218 @@ def generate_diagnostic_plot(self, ax: Optional[plt.Axes] = None) -> Optional[pl
...
+        Args:
+            ax: Optional matplotlib axes to plot on. If None, creates new figure
             
         Returns:
-            plt.Figure: Line plot of adstock decay rates
-        """
-
-        """
-            NOTE: Missing intervalType mapping in data_mapper from input collect.
+            plt.Figure if ax is None, else None
```
Comment (dhavalpatel624624) [inline python/src/robyn/visualization/pareto_visualizer.py:335]: ditto

File: python/src/robyn/visualization/pareto_visualizer.py [hunk 2]
```diff
@@ -234,23 +234,218 @@ def generate_diagnostic_plot(self, ax: Optional[plt.Axes] = None) -> Optional[pl
...
+            ax.set_xlim(0, 1)
+            
+            # Set title and labels
+            interval_type = self.mmm_data.mmmdata_spec.interval_type if self.mmm_data else "day"
+            ax.set_title('Geometric Adstock: Fixed Rate Over Time')
+            ax.set_xlabel(f'Thetas [by {interval_type}]')
+            ax.set_ylabel(None)
+            
+        elif self.adstock in [AdstockType.WEIBULL_CDF, AdstockType.WEIBULL_PDF]:
+            # Get Weibull data
```
Comment (dhavalpatel624624) [inline python/src/robyn/visualization/pareto_visualizer.py:380]: When refactoring in another PR, lets break out this buisiness logic into 2 separate functions which would plot their respective plots

**Review by dhavalpatel624624 — CHANGES_REQUESTED** [review]

File: python/src/robyn/visualization/pareto_visualizer.py [hunk 3]
```diff
@@ -234,23 +234,218 @@ def generate_diagnostic_plot(self, ax: Optional[plt.Axes] = None) -> Optional[pl
...
+        # Set up type factor levels
+        df_imme_caov['type'] = pd.Categorical(df_imme_caov['type'],
+                                            categories=['Immediate', 'Carryover'],
+                                            ordered=True)
+        
+        # Create figure if no axes provided
+        if ax is None:
+            fig, ax = plt.subplots(figsize=(10, 8))
+        else:
+            fig = None
```
[the remark this answers is not in the document]
Comment (sumalreddy17) [reply python/src/robyn/visualization/pareto_visualizer.py:259]: added return None in else loop

File: python/src/robyn/visualization/pareto_visualizer.py [hunk 4]
```diff
@@ -234,23 +234,218 @@ def generate_diagnostic_plot(self, ax: Optional[plt.Axes] = None) -> Optional[pl
             return fig
         return None
 
-    def generate_immediate_vs_carryover(self) -> plt.Figure:
+    def generate_immediate_vs_carryover(self, ax: Optional[plt.Axes] = None) -> Optional[plt.Figure]:
```
[the remark this answers is not in the document]
Comment (sumalreddy17) [reply python/src/robyn/visualization/pareto_visualizer.py:237]: no i dont see anything

**Review by dhavalpatel624624 — APPROVED** [review]

**Pull request merged**